src/validator/nodes/BaseNode.ts

```diff
@@ -5,15 +5,16 @@ export type BaseNodes = { [key: string]: BaseNode };
 export type IData = BaseNodes | BaseNode[] | string | number | boolean | undefined;
 
 export class BaseNode {
-    public type?: DataType;
-    public description?: string;
-    public data: IData;
-    public isRequired?: boolean;
-  
-    constructor(type: DataType,schema: ISchemaNode, data: IData, isRequired: boolean){
-      this.type = type;
-      this.description = schema.description;
-      this.data = data;
-      this.isRequired = isRequired;
-    }
-  }
\ No newline at end of file
+  public type?: DataType;
+  public description?: string;
+  public data: IData;
+  public isRequired?: boolean;
+  public possibleValues?: string[] | undefined;
```

Comment:
possibleValues is not relevant for all types. Why it is moved to BaseNode?

Reply:
this has changed